src/components/ha-service-picker.ts

```diff
@@ -114,11 +114,15 @@ class HaServicePicker extends LitElement {
       if (!filter) {
         return processedServices;
       }
-      return processedServices.filter(
-        (service) =>
-          service.service.toLowerCase().includes(filter) ||
-          service.name?.toLowerCase().includes(filter)
-      );
+      const split_filter = filter.split(" ");
+      return processedServices.filter((service) => {
+        const lower_service_name = service.name.toLowerCase();
+        const lower_service = service.service.toLowerCase();
+        return (
+          split_filter.every((f) => lower_service_name.includes(f)) ||
+          split_filter.every((f) => lower_service.toLowerCase().includes(f))
```

Comment:
```suggestion
          split_filter.every((f) => lower_service.includes(f))
```
You already lowercased it?

Reply:
🙏 thank you, thought I was being clever doing the lower outside the loop... not clever enough to refactor all references!